### PR TITLE
Add quic_client_stats_impl.h to QUICHE platform implementation

### PR DESCRIPTION
**Pull request by wu-bin**
*Description*:

Add quic_client_stats_impl.h to QUICHE platform implementation, with everything being no-op.

*Risk Level*: minimal: code not used yet
*Testing*: bazel test test/extensions/quic_listeners/quiche/platform:quic_platform_test --test_output=all
*Docs Changes*: none
*Release Notes*: none
[Optional Fixes #Issue]
[Optional *Deprecated*:]

### CODE REVIEW

File: test/extensions/quic_listeners/quiche/platform/quic_platform_test.cc [hunk 1]
```diff
@@ -29,6 +30,19 @@ TEST(QuicPlatformTest, QuicArraysize) {
   EXPECT_EQ(5, QUIC_ARRAYSIZE(array));
 }
 
+enum class TestEnum { ZERO = 0, ONE, TWO, COUNT };
+
+TEST(QuicPlatformTest, QuicClientStats) {
+  // Just make sure they compiles.
```
Comment (mpwarres) [inline test/extensions/quic_listeners/quiche/platform/quic_platform_test.cc]: typo: compiles -> compile

File: test/extensions/quic_listeners/quiche/platform/quic_platform_test.cc [hunk 2]
```diff
@@ -29,6 +30,19 @@ TEST(QuicPlatformTest, QuicArraysize) {
   EXPECT_EQ(5, QUIC_ARRAYSIZE(array));
 }
 
+enum class TestEnum { ZERO = 0, ONE, TWO, COUNT };
+
+TEST(QuicPlatformTest, QuicClientStats) {
+  // Just make sure they compile.
```
Comment (alyssawilk) [inline test/extensions/quic_listeners/quiche/platform/quic_platform_test.cc]: On functions where we're not implementing the actual functionality, should we have comments or a document somewhere for what expect to work and what not?

Comment (wu-bin) [reply test/extensions/quic_listeners/quiche/platform/quic_platform_test.cc]: We can 1) add comments to quic_client_stats_impl.h, 2) add a column to the [platform impl tracking spreadsheet](https://docs.google.com/spreadsheets/d/1vcd8ERD1CRSECu2gU-Xmpd6Vj0GpC8mT_-BvhU7b6EM/edit#gid=0) to say it's not actually implemented.

Or both?  WDYT?

Comment (alyssawilk) [reply test/extensions/quic_listeners/quiche/platform/quic_platform_test.cc]: I'd mildly prefer something in the code just so folks don't have to dig up the spreadsheet but as long as we're tracking it I'm good :-)

**Comment by htuch** [comment]
Please merge master to pick up https://github.com/envoyproxy/envoy/pull/5827.

File: test/extensions/quic_listeners/quiche/platform/quic_platform_test.cc (hunk 2 above)
Comment (wu-bin) [reply test/extensions/quic_listeners/quiche/platform/quic_platform_test.cc]: Added a comment in quic_client_stats_impl.h.

**Comment by wu-bin** [comment]
> Please merge master to pick up #5827.

Done.

**Pull request merged**